api/src/main/java/org/apache/iceberg/transforms/Truncate.java

```diff
@@ -394,9 +395,7 @@ public ByteBuffer apply(ByteBuffer value) {
         return null;
       }
 
-      ByteBuffer ret = value.duplicate();
-      ret.limit(Math.min(value.limit(), value.position() + length));
-      return ret;
+      return TruncateUtil.truncateByteBuffer(length, value);
```

Comment:
There's `BinaryUtil#truncateBinary(ByteBuffer, int)`, but it doesn't exactly have the same code as is used here and I think the logic is different in that it allows for 0 length.

I can try to merge the two implementations in a follow up if we want.